GitExtUtils/DebugHelpers.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System.Diagnostics;
+
+namespace GitExtUtils;
+
+/// <summary>
+///  Set of DEBUG-only helpers.
+/// </summary>
+public static class DebugHelpers
+{
+    [Conditional("DEBUG")]
+    public static void Assert(bool condition, string message)
+    {
+        if (Debugger.IsAttached)
+        {
+            Debug.Assert(condition, message);
+        }
+        else if (!condition)
+        {
+            Debugger.Launch();
+
+            if (!Debugger.IsAttached)
+            {
+                throw new InvalidOperationException(message);
+            }
+        }
```

Comment:
...the above makes me quite unhappy with this section. While nagging the dev in an invalid situation (guarded by `Debug.Fail`), it's not appropriate for a "check" situaion (i.e., `Debug.Assert`) - an assert should not bring the app down.

Reply:
That's the fundamental problem with `Debug.Assert`.
It is used for making assumptions explicit (good) - with the expectation that they will never fail.
(I see that these sanity checks are skipped in the release build for performance reasons.)
In most cases, it is wrong and unsafe to continue the execution. Sometimes, the code throws other exceptions. In many cases, it is just a lack of error handling in release builds (which is bad IMO).